extensions-core/lookups-cached-global/src/main/java/io/druid/query/lookup/namespace/NamespaceKey.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.lookup.namespace;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * NamespaceKey is used to distinguish the same mapNames of different namespace IDs
+ */
+public class NamespaceKey
```

Comment:
For thought, would it be easier/better to use a MultiKey for the composite namespace:ID key?